source/asynch_socket.c

```diff
@@ -529,8 +554,17 @@ err_t irqTCPRecv(void * arg, struct tcp_pcb * tpcb, struct pbuf * p, err_t err)
         s->event = &e;
         ((socket_api_handler_t) (s->handler))();
         s->event = NULL;
-        /* Zero the impl, since a disconnect will cause a free */
-        s->impl = NULL;
+        /* if close has been called, we have to remove impl, since it could be freed */
+        switch (tpcb->state) {
```

Comment:
This was to fix a disconnect race condition.  It's not directly related to accept_v2, but it's still an important fix.